Anchore engine and Jenkins Credentials plugin guides and snippets

bainss opened the pull request:
Some additional snippets and guides for Jenkins credentials plugin and Anchore engine using AWS and Hashicorp vault secrets backends.

knelasevero commented:
Hey @bainss! Thanks a lot for the interest in the project and for the contribution.

I am wondering if this type of guide will make sense in the docs. We added [GitOps with Flux](https://external-secrets.io/guides-gitops-using-fluxcd) guide to the docs, but it made a lot of sense since the workflow is VERY different from what we have already at the other doc guides.

But checking your examples, they are basically just creating templated secrets the same way that we already explain in normal generic guides. Would you agree?

Also, having a look over your examples, most of them did not even need to be templates, right? You could have used the data field directly setting secretKeys and remoteRef.keys, if I am not wrong.

Please let me know if I am missing something :)

bainss commented:
> Hey @bainss! Thanks a lot for the interest in the project and for the contribution.
> 
> I am wondering if this type of guide will make sense in the docs. We added [GitOps with Flux](https://external-secrets.io/guides-gitops-using-fluxcd) guide to the docs, but it made a lot of sense since the workflow is VERY different from what we have already at the other doc guides.
> 
> But checking your examples, they are basically just creating templated secrets the same way that we already explain in normal generic guides. Would you agree?
> 
> Also, having a look over your examples, most of them did not even need to be templates, right? You could have used the data field directly setting secretKeys and remoteRef.keys, if I am not wrong.
> 
> Please let me know if I am missing something :)

Hi @knelasevero ,  I think the snippets belong where they are as this is where people first go to for examples.  The templates secrets I've added offer examples for working with Anchore Engine and Jenkins Credential plugin expanding the reach of ESO to people who would otherwise maybe not consider it's use. The Hashicorp Vault examples do need to be templated (as far as I know) especially if they are used as part of a jinja2 template for helm.

knelasevero commented on docs/guides-anchore-engine-credentials.md:
```diff
@@ -0,0 +1,31 @@
+# Getting started
+
+Anchore engine is one of the best of breed security scanning used for scanning docker images, filesystems and with Harbor allowing a complete solution to container scanning at build time, nightly and with the admission controller the prevention of unscanned images from being deployed into kubernetes clusters.
```
I would rephrase this a bit:

```suggestion
Anchore Engine is an open-source project that provides a centralized service for inspection, analysis, and certification of container images. With Kubernetes, it also brings nice features like preventing unscanned images from being deployed into your clusters
```

knelasevero commented on docs/guides-anchore-engine-credentials.md:
```diff
@@ -0,0 +1,31 @@
+
+## Installing with Helm
+
+There are several parts of the installation that require credentials these being :-
```
This breaks the build of the docs. Did you build it locally?

you can build with `make docs`

and you can serve with `make serve-docs`

Then you can see the generated frontend locally on port 8000

bainss replied:
Has there been an update to the docker image as I ran this before my PR , fails now but doesn't seem related to the actual markdown


```
❯ make docs
17:52:56 [ OK ] Finished generating deepcopy and crds
/Library/Developer/CommandLineTools/usr/bin/make -C ./hack/api-docs build
docker build -t github.com/external-secrets-mkdocs:latest -f Dockerfile .
[+] Building 39.0s (9/9) FINISHED
 => [internal] load build definition from Dockerfile                                                                                                                                                                                     0.0s
 => => transferring dockerfile: 823B                                                                                                                                                                                                     0.0s
 => [internal] load .dockerignore                                                                                                                                                                                                        0.0s
 => => transferring context: 2B                                                                                                                                                                                                          0.0s
 => [internal] load metadata for docker.io/library/alpine:3.11                                                                                                                                                                           2.0s
 => [auth] library/alpine:pull token for registry-1.docker.io                                                                                                                                                                            0.0s
 => [internal] load build context                                                                                                                                                                                                        0.0s
 => => transferring context: 335B                                                                                                                                                                                                        0.0s
 => [1/3] FROM docker.io/library/alpine:3.11@sha256:bcae378eacedab83da66079d9366c8f5df542d7ed9ab23bf487e3e1a8481375d                                                                                                                     0.6s
 => => resolve docker.io/library/alpine:3.11@sha256:bcae378eacedab83da66079d9366c8f5df542d7ed9ab23bf487e3e1a8481375d                                                                                                                     0.0s
 => => sha256:bcae378eacedab83da66079d9366c8f5df542d7ed9ab23bf487e3e1a8481375d 1.64kB / 1.64kB                                                                                                                                           0.0s
 => => sha256:65d32c8f6e45a68def0d167c5751476a07710eede8d3247c968a26daa676f266 528B / 528B                                                                                                                                               0.0s
 => => sha256:1fd5850ccf13c153697413d51bb0665c03202aad5314c38ca7bf14a8250f6463 1.49kB / 1.49kB                                                                                                                                           0.0s
 => => sha256:b48a9fe99aba73065302163e59c82a1b0054810c7b9ef85eee6f1b495b162461 2.73MB / 2.73MB                                                                                                                                           0.4s
 => => extracting sha256:b48a9fe99aba73065302163e59c82a1b0054810c7b9ef85eee6f1b495b162461                                                                                                                                                0.1s
 => [2/3] COPY requirements.txt /                                                                                                                                                                                                        0.0s
 => [3/3] RUN apk add -U --no-cache     python3     python3-dev     musl-dev     bash     gcc     diffutils   && pip3 install -r /requirements.txt                                                                                      34.9s
 => exporting to image                                                                                                                                                                                                                   1.3s
 => => exporting layers                                                                                                                                                                                                                  1.3s
 => => writing image sha256:009504d5e74b2161c64c167d275097e48cd19967137d2cbd567df1c65ad6b5c7                                                                                                                                             0.0s
 => => naming to github.com/external-secrets-mkdocs:latest                                                                                                                                                                               0.0s

Use 'docker scan' to run Snyk tests against images to find vulnerabilities and learn how to fix them
./generate.sh /Users/surjitbains/Documents/git/SCM/polarpoint/a/external-secrets/hack/api-docs/../..//docs/spec.md
no required module provides package github.com/ahmetb/gen-crd-api-reference-docs; to add it:
	go get github.com/ahmetb/gen-crd-api-reference-docs
make[1]: *** [generate] Error 1
make: *** [docs] Error 2
```

knelasevero replied:
You need to get the dependency there:
`go get github.com/ahmetb/gen-crd-api-reference-docs`

Just for the make docs. We don't add it as a general dependency in the project since it is only used for generating the docs.

But make docs is failing even after adding it.

bainss replied:
```
go get github.com/ahmetb/gen-crd-api-reference-doc
go get: module github.com/ahmetb/gen-crd-api-reference-doc: git ls-remote -q origin in /Users/surjitbains/go/pkg/mod/cache/vcs/5658c250e7048acd139dcc8cab1cfef7e46665c1af2d12e2c83c240b3dbdb047: exit status 128:
	remote: Repository not found.
	fatal: repository 'https://github.com/ahmetb/gen-crd-api-reference-doc/' not found
	
```

bainss replied:
@knelasevero  What error message do you see?

knelasevero replied:
Do you mind adding me to your fork? I can take care from there :)

bainss replied:
@knelasevero done

knelasevero commented on docs/guides-anchore-engine-credentials.md:
```diff
@@ -0,0 +1,31 @@
+postgres-password
+
+
+Creating the following external secret ensure the credentials are drawn from the backend provider of choice example for Hashicorp Vault and AWS Secrets Manager providers
```
```suggestion
Creating the following external secret ensure the credentials are drawn from the backend provider of choice. The example shown here works with Hashicorp Vault and AWS Secrets Manager providers.
```
```suggestion
Creating the following external secret ensure the credentials are drawn from the backend provider of choice example for Hashicorp Vault and AWS Secrets Manager providers
```

moolen commented:
Hey @bainss, thank you for providing docs and examples for integrations. in the `guides` section we (mostly) describe features of the controller and our custom resource types. We can put your examples here but i feel that they're slightly misplaced: They provide copy-paste examples for anchore/jenkins users.

I think it makes sense to open up a new section (maybe `examples` or so?) where we store examples for different integration points like Anchore, Jenkins and also Flux.
I think it would be great to have a starting point to copy-paste yaml manifests to build a own solution. WDYT?

bainss commented:
> Hey @bainss, thank you for providing docs and examples for integrations. in the `guides` section we (mostly) describe features of the controller and our custom resource types. We can put your examples here but i feel that they're slightly misplaced: They provide copy-paste examples for anchore/jenkins users.
> 
> I think it makes sense to open up a new section (maybe `examples` or so?) where we store examples for different integration points like Anchore, Jenkins and also Flux. I think it would be great to have a starting point to copy-paste yaml manifests to build a own solution. WDYT?

Hi @moolen  yes makes complete sense to go with examples

moolen commented:
Hey @bainss do you have capacity to implement the changes? If not i can take this over

bainss commented:
> 

Hi @moolen  do you just want me to move to the examples directory ?

moolen commented:
yeah, that's it :)

knelasevero commented:
We should have a check in the ci, if changes are in the docs, to see if it builds. Last time I tried in this PR I think it could not build, if I am not mistaken

bainss commented:
@moolen  please feel free to move to the examples directory

moolen commented:
I moved the docs to the examples dir and pushed my changes to #682, closing this one.
